### DISCUSSION TITLE
[v2]: 设置面板注入与全局设置监听.

### DISCUSSION THREAD

**Discussion by FoundTheWOUT** [the1812/Bilibili-Evolved#2398] [提问 / Q&A] [answered]
在做一些功能, 有以下问题.

1. `plugin`有没有办法像给自定义顶栏注入一个按钮那样, 通过某种方法对设置(每个组件)中的选项进行信息注入. 比如, 增多一个选项.
2. 是否有方法能对全局的**任意**设置操作进行监听, 包括但不限与设置插件本身、对每个组件进行设置.

~~现在 Discussions 也能打 label 了吗, 还是说之前就可以~~

**Comment by the1812** [comment, marked as answer]
1. 往 `settingsPanel.componentActions` 注入 (`src\components\settings-panel\component-actions\component-actions.ts`)
2. `addSettingsChangeListener` ?

刚应用 Discussions 的时候没有 label, 后来出的

**Reply by the1812** [reply]
1 我也不太确定是不是你说的那个地方, 你要不截个图

**Reply by FoundTheWOUT** [reply]
![截屏2021-10-16 下午8 18 51](https://user-images.githubusercontent.com/47922588/137587097-f59dcea5-c900-4c02-9b32-961317af4358.png)
我指这样的选项

`addSettingsChangeListener`貌似也是要指定`path`的吧? 这个`path`是不是可以理解成对象(属性)的标识. 而且我注意到源码中有`addSettingsChangeListener('userComponent')`, 除了这个路径还有其它可用路径吗.

**Reply by the1812** [reply]
这个只能隐含式注入到特定组件, 界面不显示但能保存, 直接修改那个组件的 `getComponentSettings('name').options` 对象

`path` 可以套多层, 用 `.`分割, 路径可以找个页面看 `window.bilibiliEvolved.settings` 的结构

**Comment by Geeyun-JY3** [comment]
今年五月份加的：https://github.blog/changelog/2021-05-18-github-discussions-labels-and-announcements-category-format/
